backends/backends.c

```diff
@@ -313,74 +505,63 @@ void *backends_main(void *ptr) {
 
     // ------------------------------------------------------------------------
     // select the backend type
-
-    if(!strcmp(type, "graphite") || !strcmp(type, "graphite:plaintext")) {
-
-        default_port = 2003;
-        backend_response_checker = process_graphite_response;
-
-        if(BACKEND_OPTIONS_DATA_SOURCE(global_backend_options) == BACKEND_SOURCE_DATA_AS_COLLECTED)
-            backend_request_formatter = format_dimension_collected_graphite_plaintext;
-        else
-            backend_request_formatter = format_dimension_stored_graphite_plaintext;
-
-    }
-    else if(!strcmp(type, "opentsdb") || !strcmp(type, "opentsdb:telnet")) {
-
-        default_port = 4242;
-        backend_response_checker = process_opentsdb_response;
-
-        if(BACKEND_OPTIONS_DATA_SOURCE(global_backend_options) == BACKEND_SOURCE_DATA_AS_COLLECTED)
-            backend_request_formatter = format_dimension_collected_opentsdb_telnet;
-        else
-            backend_request_formatter = format_dimension_stored_opentsdb_telnet;
-
+    BACKEND_TYPE work_type = backends_select_type(type);
+    if (work_type == BACKEND_TYPE_UNKNOWN) {
+        goto cleanup;
```

Comment:
i think you need to log error here (and include `type` into it)

Reply:
Ok, I will prepare this and when Travis and LGTM finish their job, I will upload the code again with possible fix in the results of the two cited tools too.

Reply:
>  when Travis and LGTM finish their job

it will take a while :smile: 	i would not wait :smile:

Reply:
:D 
Only to complement, I saw that I already had logged inside the function "backend_select_type", but I am moving from there to the position that you specified.

Reply:
oh, sorry, you are right. But yes, log it right after `backends_select_type` is better then inside the func.

btw i would suggest ot rename

`backends_select_type` -> `backend_select_type`

Reply:
i edited the comment, see it on github

Reply:
I accepted the suggestion and I rename, but this created a problem. As you can see in this file, there are functions not created by me, that begins with "backends_", so we have to create a common pattern for the whole file. Let us wait other opinions for I change the functions and we have only one pattern.

Reply:
well, i see `backend_set_opentsdb_http_variables`... :smile: 	

Ok, if there is a pattern, then use it, i was not aware